src/main/java/com/commercetools/sync/producttypes/utils/AttributeDefinitionUpdateActionUtils.java

```diff
@@ -20,20 +29,139 @@
+            () -> ChangeIsSearchable.of(
+                oldAttributeDefinition.getName(),
+                newAttributeDefinition.isSearchable()
+            )
+        );
+    }
+
+    /**
+     * Compares the input hints of an {@link AttributeDefinition} and an {@link AttributeDefinitionDraft} and returns
+     * an {@link UpdateAction}&lt;{@link ProductType}&gt; as a result in an {@link Optional}. If both the
+     * {@link AttributeDefinition} and the {@link AttributeDefinitionDraft} have the same input hints, then no update
+     * action is needed and hence an empty {@link Optional} is returned.
+     *
+     * @param oldAttributeDefinition the attribute definition which should be updated.
+     * @param newAttributeDefinition the attribute definition draft where we get the new input hint.
+     * @return A filled optional with the update action or an empty optional if the input hints are identical.
+     */
+    @Nonnull
+    public static Optional<UpdateAction<ProductType>> buildChangeInputHintUpdateAction(
+        @Nonnull final AttributeDefinition oldAttributeDefinition,
+        @Nonnull final AttributeDefinitionDraft newAttributeDefinition) {
+
+        return buildUpdateAction(oldAttributeDefinition.getInputHint(), newAttributeDefinition.getInputHint(),
+            () -> ChangeInputHint.of(
+                oldAttributeDefinition.getName(),
+                newAttributeDefinition.getInputHint()
+            )
+        );
+    }
 
-        // TODO Add Attribute Definition Update Actions
+    /**
+     * Compares the attribute constraints of an {@link AttributeDefinition} and an {@link AttributeDefinitionDraft}
+     * and returns an {@link UpdateAction}&lt;{@link ProductType}&gt; as a result in an {@link Optional}. If both the
+     * {@link AttributeDefinition} and the {@link AttributeDefinitionDraft} have the same attribute constraints, then
+     * no update action is needed and hence an empty {@link Optional} is returned.
+     *
+     * @param oldAttributeDefinition the attribute definition which should be updated.
+     * @param newAttributeDefinition the attribute definition draft where we get the new attribute constraint.
+     * @return A filled optional with the update action or an empty optional if the attribute constraints are identical.
+     */
+    @Nonnull
+    public static Optional<UpdateAction<ProductType>> buildChangeAttributeConstraintUpdateAction(
+        @Nonnull final AttributeDefinition oldAttributeDefinition,
+        @Nonnull final AttributeDefinitionDraft newAttributeDefinition) {
 
-        return Collections.emptyList();
+        return buildUpdateAction(oldAttributeDefinition.getAttributeConstraint(),
+            newAttributeDefinition.getAttributeConstraint(),
+            () -> ChangeAttributeConstraint.of(
+                oldAttributeDefinition.getName(),
+                newAttributeDefinition.getAttributeConstraint()
+            )
+        );
     }
```

Comment:
don't forget the private empty constructor ;)

Reply:
3f1adcd1f054f72e37c1ab02e1bb844b8af45b7c